front_end/src/components/history/HistoryOverview.js

```diff
@@ -83,8 +84,11 @@ const styles = {
 
 class HistoryOverview extends Component {
   componentWillMount() {
-    this.props.getHistory();
-    this.props.getProfileInfo();
+    if (this.props.auth.isAuthenticated) {
+      // only call these if user logged in
+      this.props.getHistory();
```

Comment:
if you reuse this file, it has to be updated... it shouldn't show rewardspoints used or rewardspoints earned.